services/identity/service.go

```diff
@@ -0,0 +1,316 @@
+/*
+Identity is a service that allows storing of key/value pairs that belong to
+a given identity that is shared between multiple devices. In order to
+add/remove devices or add/remove key/value-pairs, a 'threshold' of devices
+need to vote on those changes.
+
+The key/value-pairs are stored in a personal blockchain and signed by the
+cothority using forward-links, so that an external observer can check the
+collective signatures and be assured that the blockchain is valid.
+*/
+
+package identity
+
+import (
+	"errors"
+
+	"sync"
+
+	"reflect"
+
+	"github.com/dedis/cothority/crypto"
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/network"
+	"github.com/dedis/cothority/protocols/manage"
+	"github.com/dedis/cothority/sda"
+	"github.com/dedis/cothority/services/skipchain"
+)
+
+// ServiceName can be used to refer to the name of this service
+const ServiceName = "Identity"
+
+var identityService sda.ServiceID
+
+func init() {
+	sda.RegisterNewService(ServiceName, newIdentityService)
+	identityService = sda.ServiceFactory.ServiceID(ServiceName)
+}
+
+// Service handles identities
+type Service struct {
+	*sda.ServiceProcessor
+	identities      map[string]*Storage
+	identitiesMutex sync.Mutex
+	skipchain       *skipchain.Client
+	path            string
+}
+
+// Storage stores one identity together with the skipblocks
+type Storage struct {
```

Comment:
+1 for private ! Could you also look for any other struct that is not needed outside the module ?